packages/datetime/src/timePicker.tsx

```diff
@@ -225,28 +240,27 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         );
     }
 
-    private renderDivider(text = ":") {
-        return <span className={Classes.TIMEPICKER_DIVIDER_TEXT}>{text}</span>;
-    }
-
     private renderInput(className: string, unit: TimeUnit, value: string) {
+        const { disabled, inputProps, inputComponent: CustomInputComponent } = this.props;
         const isValid = isTimeUnitValid(unit, parseInt(value, 10));
 
-        return (
-            <input
-                className={classNames(
-                    Classes.TIMEPICKER_INPUT,
-                    { [CoreClasses.intentClass(Intent.DANGER)]: !isValid },
-                    className,
-                )}
-                onBlur={this.getInputBlurHandler(unit)}
-                onChange={this.getInputChangeHandler(unit)}
-                onFocus={this.handleFocus}
-                onKeyDown={this.getInputKeyDownHandler(unit)}
-                value={value}
-                disabled={this.props.disabled}
-            />
-        );
+        const props: React.InputHTMLAttributes<HTMLInputElement> = {
+            ...inputProps,
+            className: classNames(
+                Classes.TIMEPICKER_INPUT,
+                { [CoreClasses.intentClass(Intent.DANGER)]: !isValid },
+                className,
+                inputProps ? inputProps.className : "",
+            ),
+            disabled,
+            onBlur: this.getInputBlurHandler(unit),
+            onChange: this.getInputChangeHandler(unit),
+            onFocus: this.handleFocus,
+            onKeyDown: this.getInputKeyDownHandler(unit),
+            value,
+        };
+
+        return CustomInputComponent ? <CustomInputComponent {...props} /> : <input {...props} />;
```

Comment:
I thought about this some more and I'm still not convinced or super comfortable with adding this kind of customization. We don't accept a custom component like this in any other part of our API, and I don't think #3801 presents a strong enough argument for us to start doing that here. This kind of API puts the burden on the user to properly call `onBlur`, `onChange`, etc. at the right time inside their `CustomInputComponent` and could lead to confusing bugs.